### ISSUE TITLE
[Meta] The addon load latency increased significantly in 1.1.0

### ISSUE THREAD

**Issue by ncloudioj** [mozilla/activity-stream#868]
Having been noticing this performance drop on my box since the last release, the overall performance seems to be longer than the last release. See more details [here](https://sql.telemetry.mozilla.org/dashboard/activity-stream-performance)

 let's investigate it to see if any improvement could be added.

[Edit @k88hudson]

Work being done on this includes:

- [ ] #1090 Cache the rendered add-on via page workers
- [x] #1228 URL overriding causing unnecessary perf hits 
- [ ] Remove router (needs bug)

**Comment by edmoz** [comment]
@jaredkerim can you or someone else investigate?

**Comment by edmoz** [comment]
Hmm - I'm not seeing any issues in datadog for: Sum of embedly_proxy.embedly_request_success , only  0.001% request failures for the last few days

**Comment by ncloudioj** [comment]
It doesn't appear to be an embedly proxy issue though.

As we've added a few new features in the last release, perhaps this has to do with something in there.

**Comment by rlr** [comment]
List of commits that were new in 1.1.0:
https://github.com/mozilla/activity-stream/compare/v1.0.19...v1.1.0

I started looking through the changes and nothing is sticking out as an obvious potential issue yet.

**Comment by ncloudioj** [comment]
No, it's not obvious. Let's profile it up!

**Comment by rlr** [comment]
Although, looking at the graph again... The versions are out of order so it's confusing. If you ignore 1.0.17, we get progressively slower each version from 1.0.15 to 1.1.0

**Comment by rlr** [comment]
Is it possibly that as we become more popular, more and more people on slow machines are using the addon?

**Comment by ncloudioj** [comment]
Yes, I've also noticed that. But it significantly drops more in this release.

I wonder if the way that we gauge the load latency still makes sense. As a lot of things have changed, however, the load latency measurement remains since the early day. Shall we re-evaluate it for the latest workflok?

**Comment by oyiptong** [comment]
My intuition tells me that the way we measure still makes sense.

We've been adding features and may have been slowing down rendering or whatever.

Also, getting more popular may have been adding to it too.

In the future, one way we could find out more would be to log more things to the server.

For now, we need a way to test the load latency semi or fully automatically loading different versions of the addon.

**Comment by edmoz** [comment]
action here validate or invalidate why measurement

**Comment by jaredkerim** [comment]
Moving info from @nchapman from #934 into this ticket to consolidate.



Hi all,

Let's take a look at some of the performance issues mentioned here:

https://medium.com/@manojhatax/hi-nick-e961b6f44f35#.kzu0tvn1k

        The new tab page extension to include my history and timeline was a good start. I stopped using History completely. However, items I removed from the timeline seemed to not persist across tabs.
        Adding the Experiment to my browser slowed it down noticeably. The URL bar autocomplete, in particular, was impacted to the point of becoming unusable unless I waited a few seconds after creating a new tab to start typing a URL.
        Memory usage of Firefox on my MacBook Air increased by about 15%. This observation is based on random sampling using Activity Monitor, but is a concern nonetheless.
        The Front/Back buttons stopped working correctly once a browsing session had been active for more than 8-hours. Firefox has become stable enough for me to not quit the browser instance unless I absolutely have to (typically for security updates).

    The net-net for me is that the current prototype is too immature for me to use every day. I look forward to the next release, and I hope that stability and performance issues are identified, prioritized and fixed in parallel with new feature development in the future.

**Comment by k88hudson** [comment]
Closing this in favour of using the project instead
